src/frontend/next/src/components/Posts/LoadAutoScroll.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, createRef } from 'react';
+
+import { Container, Button, Grid, createStyles, Theme } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+
+type Scroll = {
+  onScroll: Function;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    content: {
+      '& > *': {
+        padding: theme.spacing(2),
+        bottom: theme.spacing(4),
+      },
+    },
+  })
+);
+
+function LoadAutoScroll({ onScroll }: Scroll) {
+  const classes = useStyles();
+  const $buttonRef = createRef<HTMLButtonElement>();
+  // This will make the automatic infinite scrolling feature
+  // Once the "button" is on the viewport(shown on the window),
+  // The new posts are updated(call onClick() -- setSize(size + 1) in Posts.jsx --)
+  useEffect(() => {
+    const options = {
+      root: null,
+      threshold: 1.0,
+    };
+
+    const observer = new IntersectionObserver(
+      (entries) =>
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            onScroll();
+          }
+        }),
+      options
+    );
+    observer.observe($buttonRef.current!);
+
+    return () => {
+      observer.unobserve($buttonRef.current!);
```

Comment:
Need to fix the eslint warning you have on `$buttonRef.current`.  cc @birtony, who just fixed this kind of thing for #1227.